zebra-consensus/src/block/check.rs

```diff
@@ -21,14 +22,15 @@ pub fn is_coinbase_first(block: &Block) -> Result<(), BoxError> {
     let first = block
         .transactions
         .get(0)
-        .ok_or("block has no transactions")?;
+        .ok_or(TransactionError::NoTransactions)?;
```

Comment:
This is a situation where you'd want to use the `Transaction` variant on `BlockError`.
```suggestion
        .ok_or(TransactionError::NoTransactions)
        .map_err(BlockError::from)?;
```

That way the final error looks like `block contains invalid transactions: no transactions`

Reply:
Also, if you change the return type on this function from `BoxError` to `BlockError` the `map_err` I added becomes unnecessary and is implied by the `?`

Reply:
Ok, got the idea however i am not sure if it is working for me. Please check https://github.com/ZcashFoundation/zebra/pull/1070/commits/1bc2d06e1065afe2fca7dd358788a3cc8c0d81a2.
I made some local tests and the error i am getting is just the `BlockError` one, not the 2 of them, for example i triggered `CoinbasePosition` in a test case and got:

```
Error:
   0: ErrorChecker Error: block contains invalid transactions
```

Reply:
We tested this with @yaahc and it is working as we want, i was just not displaying the error correctly on my testing. So this is resolved at https://github.com/ZcashFoundation/zebra/commit/1bc2d06e1065afe2fca7dd358788a3cc8c0d81a2

Some error output correctly displayed is:

```
Error: 
   0: block contains invalid transactions
   1: first transaction must be coinbase
```